### PR TITLE
Fine-tune update policy for eslint

### LINKED ISSUE #138: Error: Environment key "es2021" is unknown
I'm using the new `es2021` environment in my `.eslintrc.json` introduced in ESLint 7.8.0:

```json
{
  "env": { "es2021": true },
  "plugins": ["decorator-position"],
  "rules": {
    "decorator-position/decorator-position": "error"
  }
}
```

And eslint-plugin-decorator-position throws the following error:

```text
Error: Error while loading rule 'decorator-position/decorator-position': .eslintrc.json » ./index.json » eslint-config-hardcore » ./base.json:
        Environment key "es2021" is unknown

Occurred while linting .eslintrc.json
    at node_modules\eslint-plugin-decorator-position\node_modules\eslint\lib\shared\config-validator.js:163:19
    at Array.forEach (<anonymous>)
    at validateEnvironment (node_modules\eslint-plugin-decorator-position\node_modules\eslint\lib\shared\config-validator.js:157:30)
    at validateConfigArray (node_modules\eslint-plugin-decorator-position\node_modules\eslint\lib\shared\config-validator.js:309:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (node_modules\eslint-plugin-decorator-position\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:464:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (node_modules\eslint-plugin-decorator-position\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:275:21)
    at CLIEngine.getConfigForFile (node_modules\eslint-plugin-decorator-position\node_modules\eslint\lib\cli-engine\cli-engine.js:933:14)
    at lineLength (node_modules\eslint-plugin-decorator-position\lib\rules\decorator-position.js:327:35)
    at Object.decoratorPositionRule [as create] (node_modules\eslint-plugin-decorator-position\lib\rules\decorator-position.js:138:27)
    at createRuleListeners (node_modules\eslint\lib\linter\linter.js:761:21)
```

eslint-plugin-decorator-position: 2.2.1
ESLint: 7.8.0

### PR DESCRIPTION
**Pull request by EvgenyOrekhov**
- Use `^` for eslint in `package.json`
- Fix dependency collision by removing eslint from devDependencies
  (since it's present in dependencies)
- Set eslint's [rangeStrategy](https://docs.renovatebot.com/configuration-options/#rangestrategy) to "bump" in `renovate.json`

These changes will prevent issues like #138 from happening in the future.

Fix #138

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,7 +64,6 @@
     "@typescript-eslint/parser": "4.3.0",
     "babel-eslint": "10.1.0",
     "common-tags": "1.8.0",
-    "eslint": "7.10.0",
```
Comment (EvgenyOrekhov) [inline package.json:67]: eslint had to be removed from devDependencies because it was causing the following yarn warning:

```
warning package.json: "dependencies" has dependency "eslint" with range "^7.10.0" that collides with a dependency in "devDependencies" of the same name with version "7.10.0"
```

**Comment by EvgenyOrekhov** [comment]
Not sure how to fix the "Lint / Commit Messages" check.

**Comment by NullVoxPopuli** [comment]
commits follow semantic commits, so for example: `fix: widen version support` and `chore: update something else that doesn't affect users`

File: package.json (hunk 1 above)
Comment (NullVoxPopuli) [reply package.json:67]: :+1: the widened version exists as a dependency, so it still works! :+1: thanks!

File: renovate.json [hunk 2]
```diff
@@ -8,5 +8,11 @@
     "smoke-tests"  
   ],
   "automerge": true,
-  "masterIssue": true
+  "masterIssue": true,
+  "packageRules": [
+    {
+      "packageNames": ["eslint"],
+      "rangeStrategy": "bump"
```
Comment (NullVoxPopuli) [inline renovate.json:15]: for documentation purposes: https://docs.renovatebot.com/configuration-options/#rangestrategy
(and historical, for looking up why this change was made)

thanks a bunch for doing this! <3

**Comment by EvgenyOrekhov** [comment]
"Lint / Commit Messages" fails for a different reason now. It says "Bad credentials": https://github.com/NullVoxPopuli/eslint-plugin-decorator-position/pull/159/checks?check_run_id=1203312253#step:5:13.

**Comment by EvgenyOrekhov** [comment]
Actually, it was failing for the same reason the first time I pushed.

**Comment by NullVoxPopuli** [comment]
eh, it's probably fine. I think GH Actions have a problem with forks. Thanks for doing this!
